KDEpy/mirroring.py

```diff
@@ -0,0 +1,136 @@
+import numpy as np
+
+
+def mirror_data(data, boundaries, pdf_values=None, decimals=10):
+    """
+    Mirrors the data based on the provided boundaries and adds up the values onto the unmirrored parts.
+    Sets everything below or above the boundaries to 0 in the values.
+
+    Parameters
+    ----------
+    data : np.ndarray
+        The input data array (N x D).
+    boundaries : list of tuples
+        A list of tuples specifying the lower and upper boundaries for each dimension.
+        Use None for no boundary in that dimension.
+        Valid examples:
+            boundaries = [(1, 10), (1, 10)]  # Both dimensions are bounded between 1 and 10
+            boundaries = [(1, 10), (None, 10)]  # Bounds (1, 10),  (-inf, 10)
+            boundaries = [(1, 10), None]     # Bounds (1, 10), (-inf, inf)
+    pdf_values : np.ndarray, optional
+        Array (N,) of the evaluated PDF at each gridpoint. If not provided, assumes uniform distribution.
+    decimals : int, optional
+        Number of decimal places to round to for grouping. Default is 10.
+
+    Returns
+    -------
+    np.ndarray
+        The mirrored data array (N x D).
+    np.ndarray
+        The rescaled PDF values.
+    """
```

Comment:
Please add a simple doctest in the docstring to showcase the function. Include some simple examples, e.g. mirroring `data = [1, 2, 2, 3, 3, 3]` across `4` and mirroring the same dataset across `[(0, 4)]`.

Showcase the use of `pdf_values` too.